recipes/medcoupling/meta.yaml

```diff
@@ -0,0 +1,75 @@
+{% set name = "medcoupling" %}
+{% set version = "9.10.0" %}
+
+
+package:
+  name: '{{ name|lower }}'
+  version: '{{ version }}'
+
+source:
+  - fn: 'medcoupling_{{ version }}.tar.gz'
+    url: https://git.salome-platform.org/gitweb/?p=tools/medcoupling.git;a=snapshot;h=V{{ version.replace('.', '_') }};sf=tgz
+    sha256: 02087a4e59ffcdab6fc68460ffd86b03a80b5c0faa695b0bfb2142f1d0d3e5ea
+    patches:
+      - swig_big_file_compile_flag.patch
+  - fn: 'config_{{ version }}.tar.gz'
+    url: https://git.salome-platform.org/gitweb/?p=tools/configuration.git;a=snapshot;h=V{{ version.replace('.', '_') }};sf=tgz
+    folder: deps/config
+    sha256: d58f6749514ce49b00590a96d6dc9090a76a07c13690e2453403643afb40346c
+
+
+build:
+  skip: True  # [osx]
+  detect_binary_files_with_prefix: true
+  # string: aster_py{{ CONDA_PY }}_mpi_{{ mpi }}_h{{ PKG_HASH }}_{{ PKG_BUILDNUM }}   # [mpi != "nompi"]
+  # string: aster_py{{ CONDA_PY }}_nompi_h{{ PKG_HASH }}_{{ PKG_BUILDNUM }}           # [mpi == "nompi"]
+  number: 0
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - make                                      # [not win]
+    - ninja                                     # [win]
+    - cmake
+    - swig >=4.0.2
+
+  host:
+    - python
+    - numpy
+    - boost
+    - zlib
+    - libxml2
+    - libmed
+    - petsc                                     # [mpi != 'nompi']
+    - {{ mpi }}                                 # [mpi != 'nompi']
+
+  run:
+    - python
+    - numpy
+    - boost
+    - libxml2
+    - zlib
+    - hdf5
+    - libmed
```

Comment:
None of these are required since they should be "exported" as runtime requirements since they were build time requirements.

Can you make a PR to `libmed` to add a `run_export`:
https://conda-forge.org/docs/maintainer/pinning_deps.html#specifying-run-exports

Then finally, we should add a global pinning to libmed by adding the current version of libmed to:
https://conda-forge.org/docs/maintainer/pinning_deps.html#updating-package-pins

Reply:
see suggestion on 54. if that doesn't clear things up, please just say so and I'll try to explain in a different way.

Reply:
Your comments on 54 helped a lot!

One final thing that I wouldn't mind clarifying regarding `run_export` with respect to which compile time dependencies libmed brings with it to another package using it.

Will adding `run_export` to libmed add the compile time requirements such as `zlib` and `hdf5` to medcoupling only because `zlib` and `hdf5` also have defined `run_export` (and not because they are `host` requirements?).

Or would adding packages without `run_export` to libmed `host` requirements also become compile time dependencies for `medcoupling`. The difference being that any updates of the packages without run_export would NOT automatically trigger a rebuild of medcoupling?

Reply:
not entirely.
see https://github.com/conda-forge/conda-forge.github.io/issues/1880#issuecomment-1408733984

Reply:
Okay, I might not fully grasp all aspects of it yet, but I think it is slowly sinking in:) Thank you for your patience!